### ISSUE TITLE
[Bug] Issue with detecting template variable dependency when query ends with variable 

### ISSUE THREAD

**Issue by cheribral** [grafana/grafana#7354]
I am using grafana version 4.1.1-1484211277 with influx on CentOS 7, and am having trouble with a nested template variable.

I have one variable called pgstat which is used in 4 other variable queries.  When I change $pgstat, 3 variables update, but one refuses to do so.  The 'stat' variable seen below will not update.  You can see the 3 top requests are updating the other variables.  The remaining two queries are for the actual graphs.  

The only difference is that $stat is a query for field keys rather than tag values.  I'm not sure how, but could this cause the problem?

![screenshot_2017-01-25_14-07-28](https://cloud.githubusercontent.com/assets/3858119/22273844/f55708a0-e307-11e6-8830-9d53892b79c6.png)

**Comment by torkelo** [comment]
Can't replicate with the exact same setup. 

![image](https://cloud.githubusercontent.com/assets/10999/22289969/99d12dca-e2fe-11e6-928e-913be80ae3a9.png)

**Comment by torkelo** [comment]
can you show the options of the stat variable?

**Comment by cheribral** [comment]
Sorry for the delay, I was away on holiday for a bit.  The difference between your setup and mine is that the stat variable isn't "show tag values", it is "show field keys".

Here is a screenshot of the options:
![screenshot_2017-02-13_17-22-18](https://cloud.githubusercontent.com/assets/3858119/22870963/67fd2ebc-f211-11e6-8dee-a246d9e1c344.png)

**Comment by torkelo** [comment]
that does not change anything :(

**Comment by cheribral** [comment]
Apparently it does though ;)

Using the latest version, if I create a brand new dashboard with only two variables:
```
$pgmetric  show measurements
$stat	   show field keys from $pgmetric
```

Nothing happens when I change pgmetric.  There is no request to the server to update the second variable.

If I change the second variable to:
`show tag values from $pgmetric with key=schema`
It makes a request every time I select something new in the first drop down.

**Comment by torkelo** [comment]
managed to replicate this issue, only occurs when the query ends with a variable. Fixed in master (and in soon to be released 4.2 beta)

**Comment by cheribral** [comment]
cool, thanks torkelo
